pkg/core/resources/model/resource.go

```diff
@@ -497,21 +497,29 @@ func ComputeLabels(r Resource, mode config_core.CpMode, isK8s bool, systemNamesp
 		}
 	}
 
-	if ns, ok := r.GetMeta().GetNameExtensions()[mesh_proto.KubeNamespaceTag]; ok && r.Descriptor().IsPolicy && r.Descriptor().IsPluginOriginated {
+	if ns, ok := r.GetMeta().GetNameExtensions()[mesh_proto.KubeNamespaceTag]; ok && r.Descriptor().IsPolicy && r.Descriptor().IsPluginOriginated && IsLocallyOriginated(mode, labels) {
 		var role mesh_proto.PolicyRole
 		switch ns {
 		case systemNamespace:
 			role = mesh_proto.SystemPolicyRole
 		default:
-			role = ComputePolicyRole(r.GetSpec().(Policy))
+			role = MustComputePolicyRole(r.GetSpec().(Policy), ns)
```

Comment:
why must is ok here?

Reply:
ah nvm, I had my own reasoning here, but it's actually easy to add `error` to `ComputeLabels()` and handle it properly. I removed `Must...` function